Auth0.Net/Client.cs

```diff
@@ -320,17 +320,29 @@ public TokenResult ExchangeAuthorizationCodePerAccessToken(string code, string r
             request.AddParameter("grant_type", "authorization_code", ParameterType.GetOrPost);
             request.AddParameter("redirect_uri", redirectUri, ParameterType.GetOrPost);
 
-            var response = this.client.Execute<Dictionary<string, string>>(request).Data;
+            var response = this.client.Execute<Dictionary<string, string>>(request);
+            var body = response.Data;

```

Comment:
This seems odd here. It might be better to access `response.Data` after you do all the checks to make sure the response is valid.